modules/javafx.graphics/src/main/java/com/sun/glass/ui/gtk/GtkView.java

```diff
@@ -113,106 +102,17 @@ protected void _uploadPixels(long ptr, Pixels pixels) {
 
     @Override
     protected void _finishInputMethodComposition(long ptr) {
-        if (imEnabled && isInPreeditMode) {
-            // Discard any pre-edited text
-            preedit.setLength(0);
-            notifyInputMethod(preedit.toString(), null, null, null, 0, 0, 0);
-        }
+        //nothing
```

Comment:
See [JDK-8320912](https://bugs.openjdk.org/browse/JDK-8320912). This routine is called when the focus moves from one TextInputControl to another. Doing nothing here leaves the previous field in a bad state.

Reply:
You're right. Fixed it.